File: Strand/BLE/PolarPMDExperiment.swift
import Foundation

/// Opt-in Polar Measurement Data (PMD) streams.
///
/// Standard 0x180D heart rate and battery remain the default path. PMD is deliberately separate
/// because keeping the ring/strap accelerometer and beat-to-beat stream active costs more battery.
/// The source reads this preference when a Polar connection is established; changing it takes effect
/// on the next connection.
enum PolarPMDExperiment {
    static let defaultsKey = "noopPolarDeepStreams"

    static var isEnabled: Bool {
        UserDefaults.standard.bool(forKey: defaultsKey)
    }
}

File: android/app/src/main/java/com/noop/ble/PolarPmdExperiment.kt
package com.noop.ble

import android.content.Context
import android.content.SharedPreferences

/**
 * Opt-in Polar Measurement Data (PMD) streams.
 *
 * Standard 0x180D heart rate and battery stay enabled regardless. PMD is a separate, default-off
 * switch because keeping Polar's accelerometer and beat-to-beat stream active costs more battery.
 * The BLE source reads it when the connection is established; changes apply on reconnect.
 */
class PolarPmdExperiment(private val prefs: SharedPreferences) {
    var enabled: Boolean
        get() = prefs.getBoolean(KEY, false)
        set(value) = prefs.edit().putBoolean(KEY, value).apply()

    companion object {
        private const val PREFS = "noop_experiments"
        const val KEY = "noopPolarDeepStreams"

        fun from(context: Context): PolarPmdExperiment =
            PolarPmdExperiment(context.getSharedPreferences(PREFS, Context.MODE_PRIVATE))
    }
}
